src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests.cs

```diff
@@ -12174,5 +12174,50 @@ public static void Main(string[] args)
                 );
             var comp = CompileAndVerify(compilation, expectedOutput: "whatever");
         }
+
+        [Fact, WorkItem(12996, "https://github.com/dotnet/roslyn/issues/12996")]
+        public void TypeOfAVarPatternVariable()
```

Comment:
I would say at least two tests must be ported to verify correctness of the changes in this PR.

---
In reply to: [75785244](https://github.com/dotnet/roslyn/pull/13274#discussion_r75785244) [](ancestors = 75785244,75774671,75774455)